### ISSUE TITLE
LoggingFailSafeMatcher IllegalStateException: Cannot resolve type description for io.opentelemetry...

### ISSUE THREAD

**Issue by pavolloffay** [open-telemetry/opentelemetry-java-instrumentation#1948]
**Describe the bug**

Agent logs 

In the Hyperace agent I see a lot of exceptions from `LoggingFailSafeMatcher IllegalStateException: Cannot resolve type description for io.opentelemetry.javaagent.tooling.<any class>`   

For instance:
```
io.opentelemetry.javaagent.tooling.bytebuddy.matcher.LoggingFailSafeMatcher - Instrumentation type matcher unexpected exception: io.opentelemetry.javaagent.instrumentation.hypertrace.java.inputstream.InputStreamInstrumentationModule
java.lang.IllegalStateException: Cannot resolve type description for io.opentelemetry.javaagent.tooling.Constant
```

This one is logged for https://github.com/hypertrace/javaagent/blob/fd1270d8c952402e572eceffd670be12815057e8/instrumentation/java-streams/src/main/java/io/opentelemetry/javaagent/instrumentation/hypertrace/java/inputstream/InputStreamInstrumentationModule.java#L70

```java
    @Override
    public ElementMatcher<? super TypeDescription> typeMatcher() {
      return extendsClass(named("java.io.InputStream"));
    }
```

@trask mentioned that similar exceptions are logged for core OTEL build and the exceptions seem harmless. Do we know the root cause of these exceptions? Are they anyhow fixable? Maybe @tylerbenson might know? 
 

```
[opentelemetry.auto.trace 2020-12-18 16:34:16:766 +0100] [main] DEBUG io.opentelemetry.javaagent.tooling.bytebuddy.matcher.LoggingFailSafeMatcher - Instrumentation type matcher unexpected exception: io.opentelemetry.javaagent.instrumentation.hypertrace.java.inputstream.InputStreamInstrumentationModule
java.lang.IllegalStateException: Cannot resolve type description for io.opentelemetry.javaagent.instrumentation.hypertrace.java.inputstream.InputStreamUtils
	at net.bytebuddy.pool.TypePool$Resolution$Illegal.resolve(TypePool.java:157)
	at io.opentelemetry.javaagent.tooling.bytebuddy.AgentCachingPoolStrategy$CachingResolution.resolve(AgentCachingPoolStrategy.java:274)
	at net.bytebuddy.pool.TypePool$Default$WithLazyResolution$LazyTypeDescription.delegate(TypePool.java:912)
	at net.bytebuddy.description.type.TypeDescription$AbstractBase$OfSimpleType$WithDelegation.getModifiers(TypeDescription.java:8383)
	at net.bytebuddy.matcher.ModifierMatcher.matches(ModifierMatcher.java:48)
	at net.bytebuddy.matcher.ModifierMatcher.matches(ModifierMatcher.java:27)
	at net.bytebuddy.matcher.NegatingMatcher.matches(NegatingMatcher.java:46)
	at net.bytebuddy.matcher.ElementMatcher$Junction$Conjunction.matches(ElementMatcher.java:122)
	at io.opentelemetry.javaagent.tooling.bytebuddy.matcher.LoggingFailSafeMatcher.matches(LoggingFailSafeMatcher.java:52)
	at net.bytebuddy.agent.builder.AgentBuilder$RawMatcher$ForElementMatchers.matches(AgentBuilder.java:1312)
	at net.bytebuddy.agent.builder.AgentBuilder$RawMatcher$Conjunction.matches(AgentBuilder.java:1163)
	at net.bytebuddy.agent.builder.AgentBuilder$RawMatcher$Conjunction.matches(AgentBuilder.java:1163)
	at net.bytebuddy.agent.builder.AgentBuilder$Default$Transformation$SimpleMatcher.matches(AgentBuilder.java:9892)
	at net.bytebuddy.agent.builder.AgentBuilder$RedefinitionStrategy$Collector.consider(AgentBuilder.java:6980)
	at net.bytebuddy.agent.builder.AgentBuilder$RedefinitionStrategy.apply(AgentBuilder.java:4834)
	at net.bytebuddy.agent.builder.AgentBuilder$Default.doInstall(AgentBuilder.java:9502)
	at net.bytebuddy.agent.builder.AgentBuilder$Default.installOn(AgentBuilder.java:9423)
	at net.bytebuddy.agent.builder.AgentBuilder$Default$Delegator.installOn(AgentBuilder.java:11025)
	at io.opentelemetry.javaagent.tooling.AgentInstaller.installBytebuddyAgent(AgentInstaller.java:143)
	at io.opentelemetry.javaagent.tooling.AgentInstaller.installBytebuddyAgent(AgentInstaller.java:68)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.opentelemetry.javaagent.bootstrap.AgentInitializer.startAgent(AgentInitializer.java:64)
	at io.opentelemetry.javaagent.bootstrap.AgentInitializer.initialize(AgentInitializer.java:50)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.opentelemetry.javaagent.OpenTelemetryAgent.agentmain(OpenTelemetryAgent.java:64)
	at io.opentelemetry.javaagent.OpenTelemetryAgent.premain(OpenTelemetryAgent.java:51)
	at org.hypertrace.agent.instrument.HypertraceAgent.agentmain(HypertraceAgent.java:68)
	at org.hypertrace.agent.instrument.HypertraceAgent.premain(HypertraceAgent.java:45)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at java.instrument/sun.instrument.InstrumentationImpl.loadClassAndStartAgent(InstrumentationImpl.java:513)
	at java.instrument/sun.instrument.InstrumentationImpl.loadClassAndCallPremain(InstrumentationImpl.java:525)
[opentelemetry.auto.trace 2020-12-18 16:34:16:770 +0100] [main] DEBUG io.opentelemetry.javaagent.tooling.bytebuddy.matcher.LoggingFailSafeMatcher - Instrumentation type matcher unexpected exception: io.opentelemetry.javaagent.instrumentation.hypertrace.java.inputstream.InputStreamInstrumentationModule
java.lang.IllegalStateException: Cannot resolve type description for io.opentelemetry.javaagent.instrumentation.javaclassloader.ClassLoaderInstrumentation$Holder
	at net.bytebuddy.pool.TypePool$Resolution$Illegal.resolve(TypePool.java:157)
	at io.opentelemetry.javaagent.tooling.bytebuddy.AgentCachingPoolStrategy$CachingResolution.resolve(AgentCachingPoolStrategy.java:274)
	at net.bytebuddy.pool.TypePool$Default$WithLazyResolution$LazyTypeDescription.delegate(TypePool.java:912)
	at net.bytebuddy.description.type.TypeDescription$AbstractBase$OfSimpleType$WithDelegation.getModifiers(TypeDescription.java:8383)
	at net.bytebuddy.matcher.ModifierMatcher.matches(ModifierMatcher.java:48)
	at net.bytebuddy.matcher.ModifierMatcher.matches(ModifierMatcher.java:27)
	at net.bytebuddy.matcher.NegatingMatcher.matches(NegatingMatcher.java:46)
	at net.bytebuddy.matcher.ElementMatcher$Junction$Conjunction.matches(ElementMatcher.java:122)
	at io.opentelemetry.javaagent.tooling.bytebuddy.matcher.LoggingFailSafeMatcher.matches(LoggingFailSafeMatcher.java:52)
	at net.bytebuddy.agent.builder.AgentBuilder$RawMatcher$ForElementMatchers.matches(AgentBuilder.java:1312)
	at net.bytebuddy.agent.builder.AgentBuilder$RawMatcher$Conjunction.matches(AgentBuilder.java:1163)
	at net.bytebuddy.agent.builder.AgentBuilder$RawMatcher$Conjunction.matches(AgentBuilder.java:1163)
	at net.bytebuddy.agent.builder.AgentBuilder$Default$Transformation$SimpleMatcher.matches(AgentBuilder.java:9892)
	at net.bytebuddy.agent.builder.AgentBuilder$RedefinitionStrategy$Collector.consider(AgentBuilder.java:6980)
	at net.bytebuddy.agent.builder.AgentBuilder$RedefinitionStrategy.apply(AgentBuilder.java:4834)
	at net.bytebuddy.agent.builder.AgentBuilder$Default.doInstall(AgentBuilder.java:9502)
	at net.bytebuddy.agent.builder.AgentBuilder$Default.installOn(AgentBuilder.java:9423)
	at net.bytebuddy.agent.builder.AgentBuilder$Default$Delegator.installOn(AgentBuilder.java:11025)
	at io.opentelemetry.javaagent.tooling.AgentInstaller.installBytebuddyAgent(AgentInstaller.java:143)
	at io.opentelemetry.javaagent.tooling.AgentInstaller.installBytebuddyAgent(AgentInstaller.java:68)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.opentelemetry.javaagent.bootstrap.AgentInitializer.startAgent(AgentInitializer.java:64)
	at io.opentelemetry.javaagent.bootstrap.AgentInitializer.initialize(AgentInitializer.java:50)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.opentelemetry.javaagent.OpenTelemetryAgent.agentmain(OpenTelemetryAgent.java:64)
	at io.opentelemetry.javaagent.OpenTelemetryAgent.premain(OpenTelemetryAgent.java:51)
	at org.hypertrace.agent.instrument.HypertraceAgent.agentmain(HypertraceAgent.java:68)
	at org.hypertrace.agent.instrument.HypertraceAgent.premain(HypertraceAgent.java:45)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at java.instrument/sun.instrument.InstrumentationImpl.loadClassAndStartAgent(InstrumentationImpl.java:513)
	at java.instrument/sun.instrument.InstrumentationImpl.loadClassAndCallPremain(InstrumentationImpl.java:525)
[opentelemetry.auto.trace 2020-12-18 16:34:16:772 +0100] [main] DEBUG io.opentelemetry.javaagent.tooling.bytebuddy.matcher.LoggingFailSafeMatcher - Instrumentation type matcher unexpected exception: io.opentelemetry.javaagent.instrumentation.hypertrace.java.inputstream.InputStreamInstrumentationModule
java.lang.IllegalStateException: Cannot resolve type description for io.opentelemetry.javaagent.tooling.Constants
	at net.bytebuddy.pool.TypePool$Resolution$Illegal.resolve(TypePool.java:157)
	at io.opentelemetry.javaagent.tooling.bytebuddy.AgentCachingPoolStrategy$CachingResolution.resolve(AgentCachingPoolStrategy.java:274)
	at net.bytebuddy.pool.TypePool$Default$WithLazyResolution$LazyTypeDescription.delegate(TypePool.java:912)
	at net.bytebuddy.description.type.TypeDescription$AbstractBase$OfSimpleType$WithDelegation.getModifiers(TypeDescription.java:8383)
	at net.bytebuddy.matcher.ModifierMatcher.matches(ModifierMatcher.java:48)
	at net.bytebuddy.matcher.ModifierMatcher.matches(ModifierMatcher.java:27)
	at net.bytebuddy.matcher.NegatingMatcher.matches(NegatingMatcher.java:46)
	at net.bytebuddy.matcher.ElementMatcher$Junction$Conjunction.matches(ElementMatcher.java:122)
	at io.opentelemetry.javaagent.tooling.bytebuddy.matcher.LoggingFailSafeMatcher.matches(LoggingFailSafeMatcher.java:52)
	at net.bytebuddy.agent.builder.AgentBuilder$RawMatcher$ForElementMatchers.matches(AgentBuilder.java:1312)
	at net.bytebuddy.agent.builder.AgentBuilder$RawMatcher$Conjunction.matches(AgentBuilder.java:1163)
	at net.bytebuddy.agent.builder.AgentBuilder$RawMatcher$Conjunction.matches(AgentBuilder.java:1163)
	at net.bytebuddy.agent.builder.AgentBuilder$Default$Transformation$SimpleMatcher.matches(AgentBuilder.java:9892)
	at net.bytebuddy.agent.builder.AgentBuilder$RedefinitionStrategy$Collector.consider(AgentBuilder.java:6980)
	at net.bytebuddy.agent.builder.AgentBuilder$RedefinitionStrategy.apply(AgentBuilder.java:4834)
	at net.bytebuddy.agent.builder.AgentBuilder$Default.doInstall(AgentBuilder.java:9502)
	at net.bytebuddy.agent.builder.AgentBuilder$Default.installOn(AgentBuilder.java:9423)
	at net.bytebuddy.agent.builder.AgentBuilder$Default$Delegator.installOn(AgentBuilder.java:11025)
	at io.opentelemetry.javaagent.tooling.AgentInstaller.installBytebuddyAgent(AgentInstaller.java:143)
	at io.opentelemetry.javaagent.tooling.AgentInstaller.installBytebuddyAgent(AgentInstaller.java:68)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.opentelemetry.javaagent.bootstrap.AgentInitializer.startAgent(AgentInitializer.java:64)
	at io.opentelemetry.javaagent.bootstrap.AgentInitializer.initialize(AgentInitializer.java:50)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.opentelemetry.javaagent.OpenTelemetryAgent.agentmain(OpenTelemetryAgent.java:64)
	at io.opentelemetry.javaagent.OpenTelemetryAgent.premain(OpenTelemetryAgent.java:51)
	at org.hypertrace.agent.instrument.HypertraceAgent.agentmain(HypertraceAgent.java:68)
	at org.hypertrace.agent.instrument.HypertraceAgent.premain(HypertraceAgent.java:45)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at java.instrument/sun.instrument.InstrumentationImpl.loadClassAndStartAgent(InstrumentationImpl.java:513)
	at java.instrument/sun.instrument.InstrumentationImpl.loadClassAndCallPremain(InstrumentationImpl.java:525)
[opentelemetry.auto.trace 2020-12-18 16:34:16:773 +0100] [main] DEBUG io.opentelemetry.javaagent.tooling.bytebuddy.matcher.LoggingFailSafeMatcher - Instrumentation type matcher unexpected exception: io.opentelemetry.javaagent.instrumentation.hypertrace.java.outputstream.OutputStreamInstrumentationModule
java.lang.IllegalStateException: Cannot resolve type description for io.opentelemetry.javaagent.tooling.Constants
	at net.bytebuddy.pool.TypePool$Resolution$Illegal.resolve(TypePool.java:157)
	at io.opentelemetry.javaagent.tooling.bytebuddy.AgentCachingPoolStrategy$CachingResolution.resolve(AgentCachingPoolStrategy.java:274)
	at net.bytebuddy.pool.TypePool$Default$WithLazyResolution$LazyTypeDescription.delegate(TypePool.java:912)
	at net.bytebuddy.description.type.TypeDescription$AbstractBase$OfSimpleType$WithDelegation.getModifiers(TypeDescription.java:8383)
	at net.bytebuddy.matcher.ModifierMatcher.matches(ModifierMatcher.java:48)
	at net.bytebuddy.matcher.ModifierMatcher.matches(ModifierMatcher.java:27)
	at net.bytebuddy.matcher.NegatingMatcher.matches(NegatingMatcher.java:46)
	at net.bytebuddy.matcher.ElementMatcher$Junction$Conjunction.matches(ElementMatcher.java:122)
	at io.opentelemetry.javaagent.tooling.bytebuddy.matcher.LoggingFailSafeMatcher.matches(LoggingFailSafeMatcher.java:52)
	at net.bytebuddy.agent.builder.AgentBuilder$RawMatcher$ForElementMatchers.matches(AgentBuilder.java:1312)
	at net.bytebuddy.agent.builder.AgentBuilder$RawMatcher$Conjunction.matches(AgentBuilder.java:1163)
	at net.bytebuddy.agent.builder.AgentBuilder$RawMatcher$Conjunction.matches(AgentBuilder.java:1163)
	at net.bytebuddy.agent.builder.AgentBuilder$Default$Transformation$SimpleMatcher.matches(AgentBuilder.java:9892)
	at net.bytebuddy.agent.builder.AgentBuilder$RedefinitionStrategy$Collector.consider(AgentBuilder.java:6980)
	at net.bytebuddy.agent.builder.AgentBuilder$RedefinitionStrategy.apply(AgentBuilder.java:4834)
	at net.bytebuddy.agent.builder.AgentBuilder$Default.doInstall(AgentBuilder.java:9502)
	at net.bytebuddy.agent.builder.AgentBuilder$Default.installOn(AgentBuilder.java:9423)
	at net.bytebuddy.agent.builder.AgentBuilder$Default$Delegator.installOn(AgentBuilder.java:11025)
	at io.opentelemetry.javaagent.tooling.AgentInstaller.installBytebuddyAgent(AgentInstaller.java:143)
	at io.opentelemetry.javaagent.tooling.AgentInstaller.installBytebuddyAgent(AgentInstaller.java:68)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.opentelemetry.javaagent.bootstrap.AgentInitializer.startAgent(AgentInitializer.java:64)
	at io.opentelemetry.javaagent.bootstrap.AgentInitializer.initialize(AgentInitializer.java:50)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.opentelemetry.javaagent.OpenTelemetryAgent.agentmain(OpenTelemetryAgent.java:64)
	at io.opentelemetry.javaagent.OpenTelemetryAgent.premain(OpenTelemetryAgent.java:51)
	at org.hypertrace.agent.instrument.HypertraceAgent.agentmain(HypertraceAgent.java:68)
	at org.hypertrace.agent.instrument.HypertraceAgent.premain(HypertraceAgent.java:45)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at java.instrument/sun.instrument.InstrumentationImpl.loadClassAndStartAgent(InstrumentationImpl.java:513)
	at java.instrument/sun.instrument.InstrumentationImpl.loadClassAndCallPremain(InstrumentationImpl.java:525)
```

[out2.txt](https://github.com/open-telemetry/opentelemetry-java-instrumentation/files/5720890/out2.txt)

**Steps to reproduce**

Run https://github.com/hypertrace/javaagent/releases on a hello world SB app.

**Comment by iNikem** [comment]
I think that #1919 addressed this?

**Comment by pavolloffay** [comment]
I will be able to confirm once a new version will be released. thanks for the pointer @iNikem

**Comment by iNikem** [comment]
@pavolloffay any news on this?

**Comment by pavolloffay** [comment]
just tested, it's fixed in 0.14. closing
